File: app/src/main/java/com/example/agroconnect/datamodel/Product.kt
package com.example.agroconnect.datamodel

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class Product(
    @field:SerializedName("estimate_exp")
    val estimateExp: String? = null,

    @field:SerializedName("amount")
    val amount: Int? = null,

    @field:SerializedName("Category")
    val category: Category? = null,

    @field:SerializedName("User")
    val user: User? = null,

    @field:SerializedName("crop_date")
    val cropDate: String? = null,

    @field:SerializedName("category_id")
    val categoryId: Int? = null,

    @field:SerializedName("user_id")
    val userId: Int? = null,

    @field:SerializedName("name")
    val name: String? = null,

    @field:SerializedName("location")
    val location: String? = null,
)

File: src/models/product.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Product extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
      Product.belongsTo(models.Category, {foreignKey:'category_id'});
      Product.belongsTo(models.User, {foreignKey:'user_id'});
    }
  }
  Product.init({
    name: DataTypes.STRING,
    amount: DataTypes.INTEGER,
    location: DataTypes.STRING,
    crop_date: DataTypes.STRING,
    estimate_exp: DataTypes.STRING,
    category_id: DataTypes.INTEGER,
    user_id: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'Product',
  });
  return Product;
};